src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,35 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                let check_lock_res =
+                    self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock));
+                if let Err(e) = check_lock_res {
+                    *can_be_cached.unwrap() = can_be_cached_by_lock;
+                    return Err(e);
+                }
             }
             IsolationLevel::Rc => {}
         }
 
-        self.load_data(user_key)
+        let mut use_near_seek = false;
+        if can_be_cached_by_lock {
```

Comment:
if my comment over L165 is applied then this line can be removed (since `can_be_cached_by_lock` are always synced to `can_be_cached`, we only need to check `can_be_cached`).